### DISCUSSION TITLE
Not execute all tasks

### DISCUSSION THREAD

**Discussion by minuth** [vercel/turborepo#1994] [Help] [answered]
I have 2 tasks to run:

- `site#dev` task
-  `server#dev` task

I want to run `server#dev` before `site#dev`. 
Here is my pipeline:
```
{
    "server#dev":{
       "cache":false
    },
    "site#dev": {
      "cache": false,
      "dependsOn": ["server#dev"]
    }
}
```

The problem is that when I run the command `yarn dev` it executes only `server#dev` task.

<img width="354" alt="image" src="https://user-images.githubusercontent.com/33181465/190866365-05ec67d1-4f70-49ac-bb29-bc48814dbeee.png">

**Comment by nathanhammond** [comment, marked as answer]
Assuming that the `server#dev` task never exits, this is expected. You would need to invoke `dev` using `--parallel`, or remove the dependency between them and let them boot up individually.

**Reply by minuth** [reply]
The `server#dev` existed and it ran successfully. My goal is not to run it parallel or individually, I want to run `server#dev` before `site#dev` with one `yarn dev` command.

**Reply by nathanhammond** [reply]
Based on the partial screenshot you've provided, `server#dev` is a long-running task. It does not appear to exit. Is that correct? If you run `yarn dev` in the `server` package, it does not exit until you send it `^C`, correct? If so, my previous answer is the solution you're looking for.

If you have two long-running tasks how would we otherwise know when it was safe to run the next one? We wait for a process to exit before proceeding with the next task, which is why you're seeing the behavior that you're seeing.

**Reply by minuth** [reply]
Thanks for the clarification, I understood now.
